flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecRank.java

```diff
@@ -37,20 +39,48 @@
 import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
 import org.apache.flink.table.types.logical.RowType;
 
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
 import java.util.Collections;
+import java.util.List;
+
+import static org.apache.flink.table.planner.plan.nodes.exec.stream.StreamExecRank.FIELD_NAME_OUTPUT_RANK_NUMBER;
 
 /**
  * {@link BatchExecNode} for Rank.
  *
  * <p>This node supports two-stage(local and global) rank to reduce data-shuffling.
  */
+@ExecNodeMetadata(
+        name = "batch-exec-rank",
```

Comment:
I cannot find this identifier in any of the attached plans. Maybe we need some additional test cases to trigger it?

Reply:
Ah, good catch.  The optimizations for rank are different in streaming and batch.  I'll go looking for a case which is optimized in batch.

Reply:
Here's the test which adds support: https://github.com/apache/flink/pull/25206/commits/ac2bd74b1e52e5c34496ef8a6f091e4100535ba6#diff-22bac638f9ee782fcf9e02e2f5cc6525def6535f4e144596fc61900779c8682fR106